src/SFML/Audio/SoundFileReaderWav.cpp

```diff
@@ -169,15 +169,29 @@ Uint64 SoundFileReaderWav::read(Int16* samples, Uint64 maxCount)
                 break;
             }
 
+            case 3:
+            {
+                Uint32 sample = 0;
+                if (decode24bit(*m_stream, sample))
+                    *samples++ = sample >> 8;
+                else
+                    return count;
+                break;
+            }
+
             case 4:
             {
-                Int32 sample = 0;
+                Uint32 sample = 0;
                 if (decode(*m_stream, sample))
                     *samples++ = sample >> 16;
                 else
                     return count;
                 break;
             }
+
+            default:
+                err() << "Failed to read from WAV sound file (unsupported sample size: " << m_bytesPerSample * 8 << " bit)" << std::endl;
+                return 0;
         }
```

Comment:
> I would remove this and put an assert instead, since the check + message is done when opening the file.

Since this is a runtime and not a logic error, I would not use assertions here. Even correct programs can encounter WAV files with unsupported sample sizes.